src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -2577,19 +2564,33 @@ public override void OnPush()
                 try
                 {
                     var task = _stage._mapFunc(message);
-                    var holder = new Holder<TOut>(message, NotYetThere, _taskCallback);
+                    var holder = new Holder<TOut>(message, NotYetThere);
                     _buffer.Enqueue(holder);
 
                     // We dispatch the task if it's ready to optimize away
                     // scheduling it to an execution context
                     if (task.IsCompleted)
                     {
-                        holder.SetElement(Result.FromTask(task));
-                        HolderCompleted(holder);
+                        HolderCompleted(holder, Result.FromTask(task));
                     }
                     else
-                        task.ContinueWith(t => holder.Invoke(Result.FromTask(t)),
-                            TaskContinuationOptions.ExecuteSynchronously);
+                    {
+                        async Task WaitForTask()
```

Comment:
Used the latest .NET meta for relying on a local function + detached `Task` + `await` on parent task instead of using `ContinueWith` - this is designed to tremendously simplify error-handling mostly.

Reply:
This has the side effect of eliminating the `AggregateException`s previously emitted from `SelectAsync`, which is why I've had to update some of the unit tests.